fdbclient/NativeAPI.actor.cpp

```diff
@@ -6990,6 +7163,16 @@ Future<Void> DatabaseContext::createSnapshot(StringRef uid, StringRef snapshot_c
 	return createSnapshotActor(this, UID::fromString(uid_str), snapshot_command);
 }
 
+DatabaseSharedState* DatabaseContext::initSharedState() {
+	DatabaseSharedState* newState = new DatabaseSharedState();
+	setSharedState(newState);
+	return newState;
+}
+
+void DatabaseContext::setSharedState(DatabaseSharedState* p) {
+	sharedStatePtr = p;
+}
+
```

Comment:
We also need logics for deleting the shared state objects. Probably, we should rely on reference counting. setSharedState would increase reference count, and the reference count could be decreased when DatabaseContext holding a reference is destroyed.